maml_rl/metalearners/maml_trpo.py

```diff
@@ -131,8 +224,18 @@ def step(self,
 
         old_loss = sum(old_losses) / num_tasks
         grads = torch.autograd.grad(old_loss,
-                                    self.policy.parameters(),
-                                    retain_graph=True)
+                                    self.agent.policy_net.parameters(),
+                                    retain_graph=True, allow_unused=True)
+        non_none_grads = []
+        non_none_indices = []
+        for i in range(len(grads)):
+            elem = grads[i]
+            if elem is not None:
+                non_none_indices.append(i)
+                non_none_grads.append(elem.contiguous())
+        grads = tuple(non_none_grads)
+        #print("Type of grads")
+        #print(grads)
```

Comment:
Is this because we have parameters that we don't update at all?

Reply:
yes!